setup.py

```diff
@@ -64,6 +64,25 @@ def finalize_options(self):
         install.finalize_options(self)
 
 
+cmdClass = {'install': CheckCairoXcb}
+
+try:
+    from sphinx.setup_command import BuildDoc
+    import subprocess
+
+    class BuildDocWrapper(BuildDoc):
+        def run(self):
+            subprocess.call(['scripts/ffibuild'])
+            subprocess.call(['make', '-C', 'docs', 'genkeyimg'])
+            super().run()
+
+    cmdClass['build_sphinx'] = BuildDocWrapper
+
+except ImportError as e:
+    print(e)
+    pass
```

Comment:
Why do we need all of this?

Reply:
this during doc buildings add image generation before. As not all have all requirements to build docs - moved under `try` block